state/machine.go

```diff
@@ -1609,14 +1609,15 @@ func (m *Machine) AddressesBySpaceID() (map[string][]corenetwork.SpaceAddress, e
 	return res, nil
 }
 
-// SubnetsBySpaceID resolves the subnet assignments for each space the machine
-// is associated with and returns them back as a map where keys are space IDs
-// and each value is a slice of subnet IDs assigned to a particular space.
-func (m *Machine) SubnetsBySpaceID() (map[string][]string, error) {
+// SubnetCIDRsBySpaceID resolves the subnet assignments for each space the
+// machine is associated with and returns them back as a map where keys are
+// space IDs and each value is a slice of subnet CIDRs assigned to a particular
+// space.
+func (m *Machine) SubnetCIDRsBySpaceID() (map[string][]string, error) {
```

Comment:
Can I get this information from SpaceInfos (or add a helper to do that?). If so, I don't mind changing my code to use SpaceInfos here and remove this method (it was added by me in the past and nothing else uses it).

Reply:
One snag here is that we want the space -> []subnet assignments for a particular machine. Is it possible for a machine to only have addresses in a subset of the subnets allocated to a space?

Reply:
On the other hand, we could grab all subnets anyway. Worst case we will generate a few bogus ingress rules (which is fine because we are not really connected to any of those subnets anyway)

Reply:
What we're actually doing here is getting all the addresses, and making a query for each to get the subnet.

If you had a materialised `SpaceInfos`, you could just call `AllSubnetInfos` and use `GetByCIDR` for each address. If you are not already in possession of a `SpaceInfos`, it might not be worth it for single-NIC machines...